lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -44,12 +44,27 @@ defmodule ExUnit.Runner do
     config = configure(opts, manager, self(), stats_pid)
     :erlang.system_flag(:backtrace_depth, Keyword.fetch!(opts, :stacktrace_depth))
 
-    start_time = System.monotonic_time()
-    EM.suite_started(config.manager, opts)
-
     modules_to_restore =
       if Keyword.fetch!(opts, :repeat_until_failure) > 0, do: {[], []}, else: nil
 
+    modules_to_restore =
+      if not opts[:dry_run] do
+        do_run(config, modules_to_restore, opts, load_us)
+      else
+        nil
+      end
+
+    stats = ExUnit.RunnerStats.stats(stats_pid)
+    EM.stop(config.manager)
+    after_suite_callbacks = Application.fetch_env!(:ex_unit, :after_suite)
+    Enum.each(after_suite_callbacks, fn callback -> callback.(stats) end)
+    {stats, modules_to_restore}
```

Comment:
@whatyouhide So, I tried to move the flag down the stack as far as possible. I still need to write tests, but does this look more in the right direction? 🙏

Reply:
I think we want to move it even farther down. Basically, everything runs as usual, but you skip running setup_all, setup and the tests themselves.

Reply:
Thank you for the direction; I've pushed some updates!